### ISSUE TITLE
last thoughts before 0.1

### ISSUE THREAD

**Issue by katieannemills** [GRIFFINCollaboration/beamCompanionExplorer#19]
All requested features are in place; will test and check Wednesday. A few outstanding points:
- [This line](https://github.com/AdamGarnsworthy/CSBBasic/blob/master/index.php#L712) always bothered me in the original CSB page; it's clearly calculating an A/Q, but the Q in the denominator is different than the Q used to determine the number of electrons to subtract from the mass.
- FYI both aluminium and stainless steel liners list the exact same constituent species. Intentional?
- Note the RFQ and DSB prebuncher resolutions reported in the text for the original version (1/1000 and 1/400 respectively) are much different than the parameters found in the code (1/400 and 1/200).
- there was a comment in the original code about the charge state width calculation being for Z>20; should this be enforced in the code? What about Z<=20?

@AdamGarnsworthy, do please have a close read of the text at the bottom when you have a chance; I get what you're doing here, but I'm not intimately familiar with your hardware, so I may have used overly generic terms to describe some parts. Also, I tried to make sure the text reflects the code faithfully, so if something seems wrong, don't overlook it as a simple textual error - it likely indicates a real mistake in the code.

**Comment by katieannemills** [comment]
Regarding the first bullet point, you can see the effect of this on for example the [50K 14+ plot](http://griffincollaboration.github.io/beamCompanionExplorer/strippingFoil.html?species=K&A=50&qOriginal=8#chargeStateOption14); there are actually two 50K 14+ in the plot; the nominally correct one at dead center, coming from selecting an 8+ state after the CSB, and one slightly to the left, coming from the analogous line to the one linked above that subtracts 14 instead of 8 electron masses (then divides by 8) when calculating the A/Q of this species after the CSB.

Should both these charges be the charge state that would allow the companion to pass through the CSB filter?

**Comment by AdamGarnsworthy** [comment]
I think its a mistake in my original code. This should be corrected.

**Comment by katieannemills** [comment]
Okay, fixed - I'm happy with this when you are, once you've had a chance to look things over, we'll stamp a release and promote it to the community.

**Comment by AdamGarnsworthy** [comment]
"Note the RFQ and DSB prebuncher resolutions reported in the text for the original version (1/1000 and 1/400 respectively) are much different than the parameters found in the code (1/400 and 1/200)."
-The theoretical resolving power of the hardware is 1/1000. It is believed this is achievable but it is very difficult for us to positively demonstrate is experimentally. It also depends on the energy spread of the beam which can change from case to case. In reality we may see a resolving power which is worse than 1/1000 and therefore a poorer resolving power is displayed to users so they are not overly optimistic in the expected cleanliness of the delivered beam. I used 1/400 but it could probably be made 1/800 and still be realistic.

**Comment by AdamGarnsworthy** [comment]
"there was a comment in the original code about the charge state width calculation being for Z>20; should this be enforced in the code? What about Z<=20?"
-Is this charge state width actually used for any further calculations once it is determined?

**Comment by AdamGarnsworthy** [comment]
"FYI both aluminium and stainless steel liners list the exact same constituent species. Intentional?"
-Intentional. When we planned to make this change from the original stainless steel liner to Aluminium we thought we would completely exclude the consistents of stainless steel from the beam. Once we did it we found these species were still present but at a much reduced intensity. Therefore these lists turn out to be the same.
